src/storage/src/source/generator.rs

```diff
@@ -245,9 +260,28 @@ fn render_simple_generator<G: Scope<Timestamp = MzOffset>>(
             return;
         };
 
-        let mut rows = generator.by_seed(mz_ore::now::SYSTEM_TIME.clone(), None, resume_offset);
-
+        let now_fn = mz_ore::now::SYSTEM_TIME.clone();
+
+        let start_instant = {
+            // We want to have our interval start at a nice round number...
+            // for example, if our tick interval is one minute, to start at a minute boundary.
+            // However, the `Interval` type from tokio can't be "floored" in that way.
+            // Instead, figure out the amount we should step back based on the wall clock,
```

Comment:
```suggestion
            // Instead, figure out the amount we should step forward based on the wall clock,
```

If I'm reading right, the code steps forwards, not backwards?